devito/petsc/logging.py

```diff
@@ -0,0 +1,133 @@
+from collections import namedtuple, OrderedDict
+from devito.petsc.iet.logging import PetscInfo
+
+
+class PetscEntry:
+    def __init__(self, **kwargs):
+        self._property_names = {k.lower(): k for k in kwargs.keys()}
```

Comment:
Unneccesary, you don't need a `dict` mapping lowercase strings to their original